nextstep-mvc/src/main/java/nextstep/mvc/DispatcherServlet.java

```diff
@@ -42,30 +39,28 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) {
         String requestUri = req.getRequestURI();
         logger.debug("Method : {}, Request URI : {}", req.getMethod(), requestUri);
 
-        HandlerMapping handler = handlerMappings.stream()
-                .filter(handlerMapping -> Objects.nonNull(handlerMapping.getHandler(req)))
-                .findFirst().orElseThrow(NotSupportedHandlerMethod::new);
-
-        HandlerExecution handlerExecution = handler.getHandler(req);
+        HandlerMapping handlerMapping = findHandlerMapping(req);
+        Object handler = handlerMapping.getHandler(req);
+        HandlerAdapter handlerAdapter = findAdapter(handler);
 
         try {
-            ModelAndView mav = handlerExecution.handle(req, resp);
+            ModelAndView mav = handlerAdapter.handle(req, resp, handler);
             mav.getView().render(mav.getModel(), req, resp);
         } catch (Exception e) {
             logger.debug("렌더링 실패");
         }
     }
 
-    private void move(String viewName, HttpServletRequest req, HttpServletResponse resp)
-            throws ServletException, IOException {
-        if (viewName.startsWith(DEFAULT_REDIRECT_PREFIX)) {
-            resp.sendRedirect(viewName.substring(DEFAULT_REDIRECT_PREFIX.length()));
-            return;
-        }
-
-        RequestDispatcher rd = req.getRequestDispatcher(viewName);
-        rd.forward(req, resp);
+    private HandlerMapping findHandlerMapping(HttpServletRequest req) {
+        return handlerMappings.stream()
+                .filter(handlerMapping -> Objects.nonNull(handlerMapping.getHandler(req)))
```

Comment:
stream API 를 사용할 때 함수형 프로그래밍의 패러다임에 맞게 작성하는 것이 좋은 방향이라 생각해요.
filter 내 로직을 두 부분으로 나눌 수 있을 것 같아요.

```suggestion
                .map(handlerMapping -> handlerMapping.getHandler(req))
                .filter(Objects::nonNull)
```

Reply:
기존 코드에서 getHandler를 두번이나 사용하고 있었다는것을 확인했습니다!  filter로직을 위와같이 수정하여 중복을 제거하였습니다!